### ISSUE TITLE
How to train the model without using apex amp (automatic mixed precision)?

### ISSUE THREAD

**Issue by MHVali** [bshall/ZeroSpeech#21]
Hi,

Thanks for providing the repository and instructions!
I was wondering if any specific problem might happen if we don't use the amp for training? Or whether I have to expect different behavior or results from the model compared to what mentioned in the paper? If yes, could you please specify?
I am trying to train the code without amp, and for that I just removed the lines which you used amp for the saving checkpoints and applying it on the encoder, decoder and the optimizer. There remains only one question in my mind!

In the code "train.py" line 99:
`torch.nn.utils.clip_grad_norm_(amp.master_params(optimizer), 1)`

I replaced "amp.master_params(optimizer)" with the whole model parameters like in the following:

`torch.nn.utils.clip_grad_norm_( chain(encoder.parameters(), decoder.parameters() ), 1)`

Is it the correct way to skip using amp for this line? Since here we apply gradient clipping to all model parameters not the master ones which amp.optimizer refers to.
